core/src/game_state/play_phase.rs

```diff
@@ -621,6 +631,35 @@ impl PlayPhase {
         Ok((InitializePhase::from_propagated(propagated), msgs))
     }
 
+    fn check_jacks_last_trick(&self) -> bool {
+        // When playing with the jacks variation, the trump number has to be Jack, so we can
+        // return early otherwise
+        if !(self.propagated.jack_variation && self.trump.number() == Some(Number::Jack)) {
+            return false
+        }
+
+        let last_trick = self.last_trick.clone().unwrap();
+        let TrickEnded {
+            winner: winner_pid,
+            ..
+        } = last_trick.complete().unwrap();
+
+        // In the jack variation, the rule can only applies if the non-landord team wins the
+        // last trick
+        if self.landlords_team.contains(&winner_pid) {
+            return false
+        }
+
+        let lt_played_cards = last_trick.played_cards();
+        let PlayedCards {
+            cards,
+            ..
+        } = lt_played_cards.iter().find(|pc| pc.id == winner_pid).unwrap();
+
+        // In the jack variation, the last trick must be won with a single (trump) jack
+        return cards.len() == 1 && cards[0].number() == Some(Number::Jack);
```

Comment:
I've actually heard of multiple variants here, FWIW.

e.g. typically it's fine to win with a pair of J (i.e. triggers J->2), but not to win with HJHJ-J (b/c the winning requires the HJ)